dyn_em/module_first_rk_step_part2.F

```diff
@@ -254,6 +256,42 @@ SUBROUTINE first_rk_step_part2 (   grid , config_flags        &
                            grid%stddev_cutoff_rand_pert,grid%gridpt_stddev_rand_pert, & 
                            grid%VERTSTRUCC,grid%VERTSTRUCS,grid%VERTAMPT       )
       ENDIF !rand_perturb_on
+
+!     If_multi_perturb: if ((config_flags%multi_perturb==1).and.(grid%id == 1) .and. &
+!         ((MOD(grid%itimestep, grid%stepsp) == 0) .or. grid%itimestep == 1)) then
+      IF ( grid%stepsp .NE. 0 ) THEN
+         stepsp_test = ( MOD(grid%itimestep, grid%stepsp) == 0 )
+      ELSE
+         stepsp_test = .FALSE.
+      END IF
```

Comment:
@pedro-jm 
Pedro,
If my modified lines are correct (lines 262-266), would you please delete lines 260-261.

Reply:
Oops, I mean the new lines are 262-268. Sorry.

Reply:
@davegill 
The recent commit has removed the comments and had enhanced the logic to handle negative values of stepsp which must be a natural number